src/code/z_debug_display.c

```diff
@@ -1,15 +1,150 @@
 #include "global.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_debug_display/func_800E9470.s")
+DebugDispObject* sDebugObjectListHead;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_debug_display/DebugDisplay_AddObject.s")
+typedef struct {
+    /* 0x00 */ s16 drawType;  // indicates which draw function to use when displaying the object
+    /* 0x04 */ void* drawArg; // segment address (display list or texture) passed to the draw funciton when called
+} DebugDispObjectInfo;        // size = 0x8
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_debug_display/DebugDisplay_DrawObjects.s")
+typedef void (*DebugDispObject_DrawFunc)(DebugDispObject*, void*, GlobalContext*);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_debug_display/func_800E95F4.s")
+void DebugDisplay_DrawSpriteI8(DebugDispObject*, void*, GlobalContext*);
+void DebugDisplay_DrawPolygon(DebugDispObject*, void*, GlobalContext*);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_debug_display/func_800E97D8.s")
+DebugDispObject* DebugDisplay_Init(void) {
+    sDebugObjectListHead = NULL;
+    return sDebugObjectListHead;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_debug_display/func_800E992C.s")
+DebugDispObject* DebugDisplay_AddObject(f32 posX, f32 posY, f32 posZ, s16 rotX, s16 rotY, s16 rotZ, f32 scaleX,
+                                        f32 scaleY, f32 scaleZ, u8 red, u8 green, u8 blue, u8 alpha, s16 type,
+                                        GraphicsContext* gfxCtx) {
+    DebugDispObject* oldHead = sDebugObjectListHead;
+
+    sDebugObjectListHead = GRAPH_ALLOC(gfxCtx, sizeof(DebugDispObject));
+    sDebugObjectListHead->pos.x = posX;
+    sDebugObjectListHead->pos.y = posY;
+    sDebugObjectListHead->pos.z = posZ;
+    sDebugObjectListHead->rot.x = rotX;
+    sDebugObjectListHead->rot.y = rotY;
+    sDebugObjectListHead->rot.z = rotZ;
+    sDebugObjectListHead->scale.x = scaleX;
+    sDebugObjectListHead->scale.y = scaleY;
+    sDebugObjectListHead->scale.z = scaleZ;
+    sDebugObjectListHead->color.r = red;
+    sDebugObjectListHead->color.g = green;
+    sDebugObjectListHead->color.b = blue;
+    sDebugObjectListHead->color.a = alpha;
+    sDebugObjectListHead->type = type;
+    sDebugObjectListHead->next = oldHead;
+    return sDebugObjectListHead;
+}
+
+#include "code/debug_display/code.c"
```

Comment:
This is a bit weird
You can put `OutName="debug_display"` in the XML. That way ZAPD will generate a `code/debug_display/debug_display.c` file instead